person_manager.rb

```diff
@@ -0,0 +1,70 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+
+class PersonManager
+  attr_accessor :people
+
+  def initialize
+    @people = []
+  end
+
+  def add_person
+    puts 'Do you want to create a student(1) or a teacher(2)? [Input the number]:'
+    person = gets.chomp.to_i
+    case person
+    when 1
+      add_student
+    when 2
+      create_teacher
+    else
+      puts "Invalid Option!\n "
+    end
+  end
```

Comment:
- [ ] The `add_person` method does not adhere to the single responsibility rule. Kindly ensure that *all* requests for input and input validation are placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.` I urge you to kindly read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)